File: ContainerWithMostWater.java
/*Given n non-negative integers a1, a2, ..., an , where each represents a point at coordinate (i, ai). n vertical lines are drawn such that the two endpoints of line i is at (i, ai) and (i, 0). Find two lines, which together with x-axis forms a container, such that the container contains the most water.*/

class Solution {
    public int maxArea(int[] height) {
        int maxarea=0;
        int low=0, high=height.length-1;
        while(low<high){
            maxarea= Math.max(maxarea, Math.min(height[low], height[high])*(high-low));
            if(height[low]<height[high])
                low++;
            else
                high--;
        }
        return maxarea;    
    }
}

File: ContainerWithMostWater.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        low = 0
        high = len(height) - 1
        maxArea = 0
        while low < high :
            width = high - low
            waterheight = height[low] if height[high] > height[low] else height[high]
            area = width*waterheight
            maxArea = maxArea if maxArea >area else area
            if (height[low] < height[high]):
                low += 1
            else:
                high-=1

        return maxArea
